### PR TITLE
docs: how to write a custom state store plugin

### PR DESCRIPTION
**Pull request by sebastian-alfers**

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/persistence/state/DurableStateExample.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+// #plugin-imports
+import akka.Done;
+import akka.persistence.state.javadsl.DurableStateUpdateStore;
+import akka.persistence.state.javadsl.GetObjectResult;
+import java.util.concurrent.CompletionStage;
+// #plugin-imports
+
+// #state-store-plugin-api
+class DurableStateExample implements DurableStateUpdateStore {
```
Comment (sebastian-alfers) [inline akka-docs/src/test/java/jdocs/persistence/state/DurableStateExample.java:15]: Made this to have some java code, as the the java api is written in scala.

File: akka-docs/src/test/scala/docs/persistence/state/PersistenceStatePluginDocSpec.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      """
+
+    val system = ActorSystem(
+      "PersistenceStatePluginDocSpec",
+      ConfigFactory
+        .parseString(providerConfig)
+      //.withFallback(ConfigFactory.parseString(PersistencePluginDocSpec.config))
+    )
+    try {
+      Persistence(system)
```
Comment (sebastian-alfers) [inline akka-docs/src/test/scala/docs/persistence/state/PersistenceStatePluginDocSpec.scala:48]: @patriknw Do you see a way to actually test if the custom plugin can be started, based on the config above? This call here seems to succeed even if I place some error like a wrong plugin class in the config above.

Comment (patriknw) [reply akka-docs/src/test/scala/docs/persistence/state/PersistenceStatePluginDocSpec.scala:48]:
```
val store = DurableStateStoreRegistry(system).durableStateStoreFor[DurableStateUpdateStore[Any]]("my-state-store")
```

**Review by patriknw — APPROVED** [review]
LGTM, thanks @sebastian-alfers

**Pull request merged**